rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java

```diff
@@ -300,4 +302,23 @@ public void peekAllFromEmpty() {
         circularBuffer.clear();
         assertTrue(circularBuffer.isEmpty());
     }
+
+    @Test
+    public void circularBufferTTL() throws InterruptedException {
+        final int ttlInSeconds = 1;
+        final RedisCircularBuffer<UUID> buffer = new RedisCircularBuffer<>(rdbi, circularBufferKey, 5, new UUIDSerialHelper(), ttlInSeconds);
+
+        buffer.clear();
+        assertTrue(buffer.isEmpty());
+
+        final UUID first = UUID.randomUUID();
+        final UUID second = UUID.randomUUID();
+        buffer.addAll(ImmutableList.of(first, second));
+        assertEquals(buffer.size(), 2);
+        assertEquals(buffer.peek(), first);
+        assertTrue(buffer.containsAll(ImmutableList.of(first, second)));
+
+        Thread.sleep(ttlInSeconds * 1000);
+        await().atMost(2, TimeUnit.SECONDS).until(buffer::isEmpty);
```

Comment:
cool but now you don't need the sleep() :)

Reply:
Whoops